packages/operation/src/evm/utils.ts

```diff
@@ -21,10 +21,17 @@ export function deriveWalletFromPath(mnemonic: string, path: string, provider: e
   return new ethers.Wallet(designatorHdNode.privateKey, provider);
 }
 
-export function getDesignatedWallet(
-  mnemonic: string,
-  requesterIndex: string,
-  provider: ethers.providers.JsonRpcProvider
-) {
-  return deriveWalletFromPath(mnemonic, `m/0/${requesterIndex}`, provider);
+export function getDesignatedWallet(mnemonic: string, requester: string, provider: ethers.providers.JsonRpcProvider) {
+  return deriveWalletFromPath(mnemonic, `m/0/${addressToDerivationPath(requester)}`, provider);
+}
+
+export function addressToDerivationPath(address: string): string {
+  const requesterBN = ethers.BigNumber.from(address);
```

Comment:
Why are we creating a big number from an address? Could you add a comment here explaining what's going on here please?

Reply:
Since we are no longer using the `requesterIndex` that used to force the requester to send a tx for creating a record on chain now we need another way for deriving the hd wallet path. 

Burak came up with a smart way of transforming the requester address to a series of ntegers that can be then be used as a derivation path.

Since by convention each section of the path can hold integers up to 2^31 we need to split the BigNumber we get from the address. Since we know that the BigNumber contains 160bits what we did was split it in 31bits chunks in order to get a path like this: m/0/1st31bits/2nd31bits/3rd31bits/4th31bits/5th31bits/6th31bits.

Please let me know if this explanation made it even more confusing than the code itself 🤭

Reply:
Ok I think that makes sense. It's worth adding an explanation like this in the code too for future readers (and future Andre who is going to forget 😄 )

Reply:
I agree 👍🏻 and I even replied on a Saturday morning because future Santiago would had have a tough time explaining it on  Monday morning 😄